composer/utils/data.py

```diff
@@ -93,26 +98,60 @@ def pil_image_collate(batch: List[Tuple[Image.Image, Union[Image.Image, Tensor]]
     return image_tensor, target_tensor
 
 
-def add_dataset_transform(dataset, transform):
-    """Flexibly add a transform to the dataset's collection of transforms.
+def add_dataset_transform(dataset: VisionDataset,
+                          transform: Callable,
+                          is_tensor_transform: bool = False) -> torch.utils.data.Dataset:
+    """Add a transform to a dataset's collection of transforms. Inserts the transform
+    before or after torchvision.transforms.ToTensor(), or at the end of the collection of
+    ToTensor() is not present.
 
     Args:
-        dataset: A torchvision-like dataset
-        transform: Function to be added to the dataset's collection of transforms
+        dataset (torch.utils.data.Dataset): A torchvision-like dataset
+        transform (Callable): Function to be added to the dataset's collection of
+            transforms
+        is_tensor_transform (bool): Whether `transform` acts on data of the type
+        torch.Tensor. If `True`, and torchvision.transforms.ToTensor() is present in
```

Comment:
For auto-highlighting of methods, use these sphinx tags like `:meth:`

The leading `~` says to only show the last part (i.e. `ToTensor`) in the docs rather than the whole name (i.e. `torchvision.transforms.ToTensor`)

```suggestion
        torch.Tensor. If ``True``, and :meth:`~torchvision.transforms.ToTensor` is present in
```

Not going to comment on all, but let's enclose these in double-ticks so they will be formatted as code. Ideally we should reference it using the `:meth:` notation but can address that in a separate PR